pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -587,6 +587,17 @@ func doSafeOpen(pathname string, base string) (int, error) {
 			return -1, fmt.Errorf("unexpected symlink found %s", currentPath)
 		}
 
+		// make sure to open directory on child filesystem such as bind mounted directory
+		if fileFmt == syscall.S_IFDIR {
+			if err = syscall.Close(childFD); err != nil {
```

Comment:
what is the purpose of this call?

Reply:
Because this child directory will be opened again below, so childFD is closed here to avoid file descriptor leak.